src/models/transaction.rs

```diff
@@ -103,55 +36,119 @@ pub(crate) fn transaction_signature_to_field_elements(transaction_signed: &Trans
+/// to be signed by the relayer before broadcasting.
+pub fn transaction_data_to_starknet_calldata(
     transaction_signed: &TransactionSigned,
-    retries: u8,
-) -> Result<Vec<Felt>, EthApiError> {
+    relayer_address: Felt,
+) -> EthApiResult<Vec<Felt>> {
     let mut signed_data = Vec::with_capacity(transaction_signed.transaction.length());
     transaction_signed.transaction.encode_without_signature(&mut signed_data);
 
+    // Extract the signature from the signed transaction
+    let mut signature = transaction_signature_to_field_elements(transaction_signed);
+
     // Pack the calldata in 31-byte chunks
     let mut signed_data: Vec<Felt> = std::iter::once(Felt::from(signed_data.len()))
         .chain(signed_data.chunks(31).map(Felt::from_bytes_be_slice))
         .collect();
 
     // Prepare the calldata for the Starknet invoke transaction
-    let capacity = 6 + signed_data.len();
+    let capacity = 10 + signed_data.len() + signature.len() + 1;
 
     // Check if call data is too large
     #[cfg(not(feature = "hive"))]
     if capacity > *MAX_FELTS_IN_CALLDATA {
         return Err(EthApiError::CalldataExceededLimit(*MAX_FELTS_IN_CALLDATA, capacity));
     }
 
-    let mut calldata = Vec::with_capacity(capacity);
-
-    // assert that the selector < Felt::MAX - retries
-    assert!(*ETH_SEND_TRANSACTION < Felt::MAX - Felt::from(retries));
-    let selector = *ETH_SEND_TRANSACTION + Felt::from(retries);
-
-    // Retries are used to alter the transaction hash in order to avoid the
-    // `DuplicateTx` error from the Starknet gateway, encountered whenever
-    // a transaction with the same hash is sent multiple times.
-    // We add the retries to the selector in the calldata, since the selector
-    // is not used by the EOA contract during the transaction execution.
-    calldata.append(&mut vec![
-        Felt::ONE,                // call array length
-        *KAKAROT_ADDRESS,         // contract address
-        selector,                 // selector + retries
-        Felt::ZERO,               // data offset
-        signed_data.len().into(), // data length
-        signed_data.len().into(), // calldata length
+    let mut execute_from_outside_calldata = Vec::with_capacity(capacity);
+
+    // Construct the execute from outside calldata
+    execute_from_outside_calldata.append(&mut vec![
+        relayer_address,          // OutsideExecution caller
+        Felt::ZERO,               // OutsideExecution nonce
+        Felt::ZERO,               // OutsideExecution execute_after
+        Felt::MAX,                // OutsideExecution execute_before
+        Felt::ONE,                // call_array_len
+        *KAKAROT_ADDRESS,         // CallArray to
+        *ETH_SEND_TRANSACTION,    // CallArray selector
+        Felt::ZERO,               // CallArray data_offset
+        signed_data.len().into(), // CallArray data_len
+        signed_data.len().into(), // calldata_len
     ]);
```

Comment:
Maybe if possible, we should add the link (I imagine the source is the Kakarot Cairo code or the SNIP) to see where this order comes from, right? So that we don't get lost in the future and a reader can easily understand and not want to change this order?